composer/profiler/json_trace.py

```diff
@@ -141,18 +144,17 @@ def process_duration_event(
         name: str,
         categories: Union[List[str], Tuple[str, ...]],
         is_start: bool,
-        epoch: Optional[int],
-        step: Optional[int],
+        timestamp: Optional[Timestamp],
         wall_clock_time_ns: int,
         process_id: int,
         thread_id: int,
     ) -> None:
         ph = "B" if is_start else "E"
         args = {}
-        if epoch is not None:
-            args["epoch"] = epoch
-        if step is not None:
-            args["step"] = step
+        if timestamp.epoch is not None:
```

Comment:
If the timestamp is required, then this check can be removed. Not sure why we made it optional in the first place since it's required in the base class...